HeterogeneousCore/SonicCore/interface/RetrySameServerAction.h

```diff
@@ -0,0 +1,16 @@
+#include "HeterogeneousCore/SonicCore/interface/RetryActionBase.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicClientBase.h"
+
+class RetrySameServerAction : public RetryActionBase {
+public:
+  RetrySameServerAction(const edm::ParameterSet& pset, SonicClientBase* client)
+      : RetryActionBase(pset, client), allowedTries_(pset.getUntrackedParameter<unsigned>("allowedTries", 0)) {}
+
+  void start() override { tries_=0;};
```

Comment:
need to apply `scram b code-format` before submitting the PR upstream (can just do this at the end to avoid multiple interspersed formatting commits)